source/server/configuration_impl.h

```diff
@@ -120,6 +121,7 @@ class MainImpl : Logger::Loggable<Logger::Id::config>, public Main {
   std::unique_ptr<Upstream::ClusterManager> cluster_manager_;
   std::list<Stats::SinkPtr> stats_sinks_;
   std::chrono::milliseconds stats_flush_interval_;
+  bool stats_flush_on_admin_;
```

Comment:
it doesn't look like this gets initialized on all paths. Maybe declare it as
```
stats_flush_on_admin_{false}
```
to be sure?